dashboard/lib/dynamic_config/datastore_cache.rb

```diff
@@ -0,0 +1,72 @@
+# A caching layer that sits in front of a datastore that
+# implements get and set
+
+class DatastoreCache
+  # @param datastore [Object] a datastore adapter
+  # @param cache_expiration [int] seconds after which a cached entry expires
+  def initialize(datastore, cache_expiration: 30)
+    @cache = {}
+    @datastore = datastore
+    @cache_expiration = cache_expiration
+
+    # Note we intentionally do this before spawning the background thread
+    # to make sure the cache is seeded successfully on init
+    seed_cache
```

Comment:
Note this initializing the Datastore will blow up with an exception (via the call to all) if DynamoDB is unavailable.